src/client/eppo-client.spec.ts

```diff
@@ -518,12 +520,16 @@ describe('EppoClient E2E test', () => {
       client.getStringAssignment(flagKey, 'subject-10', {}, 'default'); // log this assignment
       client.getStringAssignment(flagKey, 'subject-10', {}, 'default'); // cache hit, don't log
 
+      expect(td.explain(mockLogger.logAssignment).callCount).toEqual(2);
```

Comment:
@aarsilv @leoromanovsky this new Set-based assignment cache we discussed breaks this test. We make an assumption here that changing the variation key will cause `logAssignment` to called again, but that's no longer the case since we removed the logic from `AbstractAssignmentCache#has`
WDYT?

Reply:
I think to decide if something has been logged, we need to factor in (hash):
* subject key
* flag key
* allocation key
* variation key

The hash of these four things is what ends up in our Set, so if the variation key changes, it should indeed cause logging to be called again.

Reply:
ah just went through with the debugger here, the bug was not obvious but yes there's a bug.
if we change the variation and log again, we'll log (that part is working), but when we revert back to the original variation key in this test, the old cache key is still there so it's not logged. maybe that's why we hadn't included the variation key as part of the cache key in the first place, because of this use case?

Reply:
Ha, yeah, I was just looking at your other changes, and it all seems sound! This is an interesting use case: do we need to log assignments if they've changed? Or can we blindly dedupe? I think--at least for now--we proceed with the latter for the reason that it's simpler and flip-flopping assignments shouldn't happen much (at least until switchback testing?), and if they do, our analysis pipeline will throw that person out anyways 🤷‍♂️

Reply:
ok I'll move forward with the assumption that we don't need to re-log then, will update the test to reflect that!